### PR TITLE
feature: allow pagination when listing prs

### PR DESCRIPTION
**Pull request by pedrocodacy**
this will enable us to get single pages of pull requests instead of iterating through all the pages to get in bulk

### CODE REVIEW

File: src/main/scala/com/codacy/client/stash/service/PullRequestServices.scala [hunk 1]
```diff
@@ -21,13 +21,21 @@ class PullRequestServices(client: StashClient) {
       direction: String = "INCOMING",
       state: String = "OPEN",
       order: String = "NEWEST",
-      includeAvatar: Boolean = false
+      includeAvatar: Boolean = false,
+      pageRequest: Option[PageRequest]
```
Comment (lolgab) [inline src/main/scala/com/codacy/client/stash/service/PullRequestServices.scala:25]: Can you have non optional parameters after parameters with default values?
Isn't it better if the mandatory parameters come first?
Also keep in mind that this change breaks binary compatibility, so add a `:breaking:`

**Review by pedrobpereira — APPROVED** [review]

File: src/main/scala/com/codacy/client/stash/service/PullRequestServices.scala (hunk 1 above)
Comment (vascorsd) [reply src/main/scala/com/codacy/client/stash/service/PullRequestServices.scala:25]: Mandatory things should be before parameters with default values, if not the point of using default values kinda disappears a bit.

**Review by lolgab — APPROVED** [review]
Have you considered to add an overload that recieves a `PageRequest` instead of a single method receiving `Option[PageRequest]`? Or the point is to force people to update and move to the paginated version?
Anyway I leave the decision to you and I approve in the meantime.

**Comment by pedrocodacy** [comment]
> Have you considered to add an overload that recieves a `PageRequest` instead of a single method receiving `Option[PageRequest]`? Or the point is to force people to update and move to the paginated version? Anyway I leave the decision to you and I approve in the meantime.

Its the second one. We should not be using this operation without pagination, its just a risk of blowing up stuff with timeouts and maybe even filling up the memory of who is using this lib (while leaving the opening for it to be set as None to not block any development that may be needed here in the meanwhile)

**Comment by lolgab** [comment]
Then probably a good option is as I suggested in the companion PR for bitbucket-scala-client.
You deprecate the old signature with a message to use the paginated one, you make the PR non breaking and when all the users have updated you can remove the deprecated version.

**Comment by pedrocodacy** [comment]
I like that, will change

**Comment by machadoit** [comment]
Looks good to me as it is, and Bitbucket Server is dying either way

**Review by machadoit — APPROVED** [review]

**Pull request merged**